src/QsCompiler/CompilerOptimizations/SideEffectChecking.fs

```diff
@@ -8,24 +8,36 @@ open Microsoft.Quantum.QsCompiler.Transformations.Core
 open Utils
 
 
+/// A ScopeTransformation that tracks what side effects the transformed code could cause
```

Comment:
Could you be more specific regarding what you mean by side effects? Thanks!

Reply:
The transformation is meant to track any kind of side effect we might have. Documentation on the functions in the class specify what side effects it currently looks for.